### PR TITLE
DOC/TEST: Make doctest work

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

File: ibis/expr/api.py [hunk 1]
```diff
@@ -1773,29 +1773,73 @@ def join(left, right, predicates=(), how='inner'):
...
+              left:
+                CrossJoin[table]
+                  left:
+                    Table: ref_1
+                  right:
+                    Table: ref_2
+              right:
+                Table: ref_3
+          right:
+            Table: ref_4
```
Comment (wesm) [inline ibis/expr/api.py]: I have to admit that I don't like doctests, but this is OK for now =)

File: ibis/impala/api.py [hunk 2]
```diff
@@ -48,37 +48,49 @@ def connect(host='localhost', port=21050, database='default', timeout=45,
...
     --------
-    >>> hdfs = ibis.hdfs_connect(**hdfs_params)
-    >>> client = ibis.impala.connect(hdfs_client=hdfs, **impala_params)
+    >>> import ibis
+    >>> hdfs = ibis.hdfs_connect(host='impala', port=50070)
+    >>> hdfs  # doctest: +ELLIPSIS
+    <ibis.filesystems.WebHDFS object at 0x...>
+    >>> client = ibis.impala.connect(hdfs_client=hdfs, port=21050)
+    >>> client  # doctest: +ELLIPSIS
+    <ibis.impala.client.ImpalaClient object at 0x...>
```
Comment (wesm) [inline ibis/impala/api.py]: do these need to be doctests?

Comment (cpcloud) [inline ibis/impala/api.py]: No, but I like having examples that actually work for users so they don't have to go spelunking around the test suite :)

Comment (wesm) [inline ibis/impala/api.py]: Agreed; I guess doctests keep us honest that the examples in the docstrings still work

**Comment by cpcloud** [comment]
Merging on green.

**Pull request closed without merging**